packages/mdc-data-table/README.md

```diff
@@ -173,3 +187,66 @@ Mixin | Description
 ## Accessibility
 
 Please refer [WAI-ARIA Authoring Practices for table](https://www.w3.org/TR/wai-aria-practices-1.1/#table) for ARIA recommended role, states & properties required for table element.
+
+## Events
+
+Please refer MDCDataTable's constant file to access these event constants.
+
+```ts
+const {events} from '@material/data-table/constants';
+// `events.ROW_SELECTION_CHANGED` to access event constant.
+```
+
+Event constant | Event name | Description
+-- | -- | --
+`ROW_SELECTION_CHANGED` | `MDCDataTable:changed` | Event emitted when row checkbox is checked or unchecked.
+`SELECTED_ALL` | `MDCDataTable:selectedAll` | Event emitted when header row checkbox is checked.
+`UNSELECTED_ALL` | `MDCDataTable:unselectedAll` | Event emitted when header row checkbox is unchecked.
+
+## `MDCDataTable` Properties and Methods
+
+Method Signature | Description
+--- | ---
+`layout() => void` | Registers new row checkbox, header row checkbox and updates the state of header row checkbox. Use this when rows are added / removed from data table.
+`getRows() => HTMLElement[]` | Returns array of row elements.
+`getSelectedRowIds() => Array<string|nul>` | Returns array of selected row ids.
+`setSelectedRowIds(rowIds: string[])` | Sets selected row ids. Overwrites previously selected rows.
+
+## Usage within Web Frameworks
+
+If you are using a JavaScript framework, such as React or Angular, you can create a Dialog for your framework. Depending on your needs, you can use the _Simple Approach: Wrapping MDC Web Vanilla Components_, or the _Advanced Approach: Using Foundations and Adapters_. Please follow the instructions [here](../../docs/integrating-into-frameworks.md).
+
+### `MDCDataTableAdapter`
+
+Method Signature | Description
+--- | ---
+`addClassAtRowIndex(rowIndex: number, cssClasses: string) => void` | Adds a class name to row element at given row index excluding header row.
+`getAttributeAtRowIndex(rowIndex: number, attr: string) => void` | Returns attribute value of row element at given row index excluding header row.
+`getRowCount() => number` | Returns row count excluding header row.
+`getRowElements() => HTMLElement[]` | Returns array of row elements excluding header row.
+`getRowIdAtIndex(rowIndex: number) => string | null` | Returns row id of row element at given row index based on `data-row-id` attribute on row element `tr`.
+`getRowIndexByChildElement(el: Element) => number` | Returns index of row element that contains give child element.
+`getSelectedRowCount() => number` | Returns selected row count.
+`isHeaderRowCheckboxChecked() => boolean` | Returns true if header row checkbox is checked.
+`isRowsSelectable() => boolean` | Returns true if table rows are selectable.
```

Comment:
`...if all table rows...`

Reply:
Table is not supposed to have some selectable rows. I think this implies that all rows are selectable when at least one row is selectable.

Reply:
Maybe it should say `Returns true if at least one row is selectable.`